### DISCUSSION TITLE
Controlling `:autoport` formatting

### DISCUSSION THREAD

**Discussion by KaranAhlawat** [joaotavora/eglot#1313] [Q&A] [answered]
I was configuring an LSP that I want to start on an automatic port. I saw the `:autoport` option in the manual, and tried to use that.
The issue I ran into was that the string into which the value of the port is substituted is like this
```
(... "--start-java-language-server=listen:" :autoport)
```
Due to which, the string after substitution is
```
"--start-java-language-server=listen: 43290"
```
which causes the LSP to not parse the port and instead throw an error.

Is there some way I can control that?

**Comment by KaranAhlawat** [comment]
By control I mean I'd like it to be of the form

```
"--start-java-language-server=listen:43290"
```

**Reply by joaotavora** [reply, marked as answer]
I don't think the `:autoport` functionality allows that, sorry.  You'd have to change whatever program you're trying to invoke to accept option values in the usual `getopt` manner.  Or create a wrapper script and customize Eglot to use that.

**Reply by KaranAhlawat** [reply]
Okay! I think the script route sounds the best, thank you.
